File: server/models/forms/personal_forms/credit_info.go
package personal_forms

type LoanDetails struct {
	LoanType           string  `form:"loanType" json:"loan_type"`
	Lender             string  `form:"lender" json:"lender"`
	SanctionedAmount   string  `form:"sanctionedAmount" json:"sanctioned_amount"`
	OutstandingAmount  string  `form:"outstandingAmount" json:"outstanding_amount"`
	EmiAmount          string  `form:"emiAmount" json:"emi_amount"`
}

type UserCreditInfo struct {
	UserID  string        `json:"user_id"`
	Loans   []LoanDetails `json:"loans"`
}

File: aarthik_setu/lib/models/profiles/personal_profile/components/credit_info.dart
class CreditInfo {
  String loanType;
  String lender;
  double sanctionedAmount;
  double outstandingAmount;
  double emiAmount;

  CreditInfo({
    required this.loanType,
    required this.lender,
    required this.sanctionedAmount,
    required this.outstandingAmount,
    required this.emiAmount,
  });

  Map<String, dynamic> toJson() {
    return {
      'loanType': loanType,
      'lender': lender,
      'sanctionedAmount': sanctionedAmount,
      'outstandingAmount': outstandingAmount,
      'emiAmount': emiAmount,
    };
  }
}